src/logic/scripting/lua/libs/libworld.cpp

```diff
@@ -115,6 +119,76 @@ static int l_get_generator(lua::State* L) {
     return lua::pushstring(L, require_world_info().generator);
 }
 
+static int l_get_chunk_data(lua::State* L) {
+    int x = (int)lua::tointeger(L, 1);
+    int y = (int)lua::tointeger(L, 2);
+    const auto& chunk = level->chunks->getChunk(x, y);
+    if (chunk->isEmpty()) {
+        return 0;
+    }
+
+    bool compress = false;
+    if (lua::gettop(L) >= 3) {
+        compress = lua::toboolean(L, 3);
+    }
+
+    if (compress) {
+        return lua::newuserdata<lua::LuaBytearray>(

```

Comment:
tests i've made for my implementation show that gzip over extrle16 is the best combination for chunks:

| average        | gzip | extrle16 | extrle16 + gzip |
| -------------- | ---- | -------- | --------------- |
| time (mcs)     | 1196 | 78       | 176             |
| output (bytes) | 1476 | 2642     | 1011            |

Reply:
Good to know, thank you